content/typeclasses-syntax.md

```diff
@@ -0,0 +1,373 @@
+
+### 5. Cleanup of Given Syntax
+
+A good language syntax is like a Bach fugue: A small set of motifs is combined in a multitude of harmonic ways. Dissonances and irregularities should be avoided.
+
+When designing Scala 3, I believe that, by and large, we achieved that goal, except in one area, which is the syntax of givens. There _are_ some glaring dissonances, as seen in this code for defining an ordering on lists:
+```scala
+given [A](using Ord[A]): Ord[List[A]] with
+  def compare(x: List[A], y: List[A]) = ...
+```
+The `:` feels utterly foreign in this position. It's definitely not a type ascription, so what is its role? Just as bad is the trailing `with`. Everywhere else we use braces or trailing `:` to start a scope of nested definitions, so the need of `with` sticks out like a sore thumb.
+
+We arrived at that syntax not because of a flight of fancy but because even after trying for about a year to find other solutions it seemed like the least bad alternative. The awkwardness of the given syntax arose because we insisted that givens could be named or anonymous, with the default on anonymous, that we would not use underscore for an anonymous given, and that the name, if present, had to come first, and have the form `name [parameters] :`. In retrospect, that last requirement showed a lack of creativity on our part.
+
+Sometimes unconventional syntax grows on you and becomes natural after a while. But here it was unfortunately the opposite. The longer I used given definitions in this style the more awkward they felt, in particular since the rest of the language seemed so much better put together by comparison. And I believe many others agree with me on this. Since the current syntax is unnatural and esoteric, this means it's difficult to discover and very foreign even after that. This makes it much harder to learn and apply givens than it need be.
+
+**Proposal:** Things become much simpler if we introduce the optional name instead with an `as name` clause at the end, just like we did for context bounds. We can then use a more intuitive syntax for givens like this:
+```scala
+given Ord[String]:
+  def compare(x: String, y: String) = ...
+
+given [A : Ord] => Ord[List[A]]:
+  def compare(x: List[A], y: List[A]) = ...
+
+given Monoid[Int]:
+  extension (x: Int) def combine(y: Int) = x + y
+  def unit = 0
+```
+If explicit names are desired, we add them with `as` clauses:
+```scala
+given Ord[String] as stringOrd:
+  def compare(x: String, y: String) = ...
+
+given [A : Ord] => Ord[List[A]] as listOrd:
+  def compare(x: List[A], y: List[A]) = ...
+
+given Monoid[Int] as intMonoid:
+  extension (x: Int) def combine(y: Int) = x + y
+  def unit = 0
+```
+
+The underlying principles are:
+
+ - A `given` clause consists of the following elements:
+
+    - An optional _precondition_, which introduces type parameters and/or using clauses and which ends in `=>`,
+    - the implemented _type_,
+    - an optional name binding using `as`,
```

Comment:
Perhaps the meta-rule here is:

1. `<prefix> as <name>` with the name on the right is used where the name is usually optional (import renames, pattern-match name bindings, given names)

2. `<keyword> <name> <suffix>` style (`val`, `def`, `object`, `class`, etc.) with the name on the left is used where the name is mandatory and important